MapQuest/iso-3166
# build: move @mapquest/team to `optionalDependencies`

Moving internal tooling dependency to optionalDependencies to prevent installation failures for public users while maintaining internal development workflows.

**Jira ticket number:** N/A - Public GitHub contribution

**Does this PR introduce a breaking change?** No

**Are all of the original requirements satisfied in this PR?** Yes

## What is the current behavior?
Currently, attempting to install iso-3166 from GitHub fails because npm/yarn try to fetch the @mapquest/team package from GitHub's package registry, which requires authentication. This prevents public users from being able to install and use the package.

Error message example:
```
npm error code E401
npm error 401 Unauthorized - GET https://npm.pkg.github.com/download/@mapquest/team/0.3.8/6a517b00cd28546eae76bd98c593b263da17dae2 - authentication token not provided
```

## What is the new behavior?
- Package can be installed by public users without authentication errors
- Internal MapQuest development workflows remain unchanged
- @mapquest/team package is now optional, allowing the installation to succeed even when the package cannot be accessed

## How to test
1. Clone this PR's branch
2. Run `yarn add .` or `npm install .` in a clean directory
3. Verify the installation succeeds without authentication errors
4. For MapQuest team members: verify that eslint and jest configurations still work as expected when authenticated

## Motivation and context
The package is MIT licensed and publicly available on GitHub, but the current dependency structure prevents public users from installing it directly. This change maintains all internal tooling while making the package more accessible to the open source community.

## Checklist
- [x] The title of this PR follows the conventional commit guidelines
- [x] Acceptance criteria from the original issue has been satisfied
- [x] I have performed a self-review of my code and the expected functionality
- [x] Changes are minimal and focused on solving the installation issue
- [x] No documentation changes needed as this is an internal configuration change
- [x] The changes introduced can go live immediately
- [ ] Product or Business stakeholders have performed acceptance testing
